discord-scripts/invite-management.ts

```diff
@@ -109,71 +69,75 @@ export default async function sendInvite(discordClient: Client, robot: Robot) {
-        const regex = /^ext-.*-audit$/
-        if (regex.test(channel.name)) {
-          try {
-            const channelId = channel.id
-            const maxAge = 86400 // 1 day
-            const maxUses = 5
-            const inviteUrl = await createInvite(
-              discordClient,
-              channelId,
-              maxAge,
-              maxUses,
-            )
-            if (inviteUrl) {
-              robot.logger.info(
-                `New invite created for defense audit channel: ${channel.name}, URL: ${inviteUrl}`,
-              )
-              if (channel instanceof TextChannel) {
-                channel.send(
-                  `Here is your invite link: ${inviteUrl}\n` +
-                    `This invite expires in ${
-                      maxAge / 3600
-                    } hours and has a maximum of ${maxUses} uses.`,
-                )
-              }
-            }
-            // create a new role with the exact name of the channel with permissions only to that channel
-            const role = await channel.guild.roles.create({
-              name: channel.name,
-              reason: `Role for ${channel.name} channel`,
-            })
-
-            await channel.permissionOverwrites.create(role, {
-              ViewChannel: true,
-              SendMessages: true,
-              ReadMessageHistory: true,
-            })
-            if (channel instanceof TextChannel) {
-              channel.send(
-                `Role ${role.name} created and permissions set for ${channel.name}`,
-              )
-            }
+      if (
+        channel.parent &&
+        channel.parent.name === "defense" &&
+        channel instanceof TextChannel &&
+        EXTERNAL_AUDIT_CHANNEL_REGEXP.test(channel.name)
+      ) {
+        try {
+          const channelId = channel.id
+          const maxAge = WEEK
+          const maxUses = 10
+          const inviteUrl = await createInvite(
+            discordClient,
+            channelId,
+            maxAge,
+            maxUses,
+          )
+          if (inviteUrl) {
             robot.logger.info(
-              `Role ${role.name} created and permissions set for channel ${channel.name}`,
+              `New invite created for defense audit channel: ${channel.name}, URL: ${inviteUrl}`,
             )
-          } catch (error) {
-            robot.logger.error(
-              `An error occurred setting up the defense audit channel: ${error}`,
+            channel.send(
+              `Here is your invite link: ${inviteUrl}\n` +
+                `This invite expires in ${
+                  maxAge / (3600 * 24)
+                } days and has a maximum of ${maxUses} uses.`,
             )
           }
+          // Create a new role with the client name extracted and set permissions to that channel
+          const auditChannel = channel.name.split("-")
+          const clientName = auditChannel[0].includes("🔒")
+            ? auditChannel[1]
+            : ""
```

Comment:
This feels like it would fall over if e.g. we had a multi-word client name. So for example `ext-tally-ho-audit` would probably result in `Defense: tally` instead of `Defense: Tally Ho` (the ideal outcome).

Let me take the `split` approach one step further and suggest:

```suggestion
          // Create a new role with the client name extracted and set permissions to that channel
          const clientName =
            channel.name
              .split("-")
              // Drop the first and last segments.
              .slice(1, -1)
              // Capitalize each segment.
              .map((segment) => segment.substring(0,1).toUpperCase() + segment.substring(1))
              // Treat the dash separators as spaces.
              .join(" ")
```

In this scenario, I would also check to make sure the `clientName` isn't empty, and if it is just log an error with the channel name and skip the role creation.